ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1892,6 +1956,62 @@ void addChunkId(StoreKey storeKey, long chunkSize, int chunkIndex) {
       indexToChunkIdsAndChunkSizes.put(chunkIndex, new Pair<>(storeKey, chunkSize));
     }
 
+    /**
+     * Validate the reserved metadata chunk id is passed in correctly where it needs to be specified.
+     * @throws RouterException in case of any errors.
+     */
+    private void validateReservedMetadataChunkId() throws RouterException {
+      if (!options.isChunkUpload() && Objects.nonNull(passedInBlobProperties.getReservedMetadataBlobId())) {
+        reservedMetadataIdMetrics.reservedMetadataPassedInForNonChunkedUploadCount.inc();
+        throw new RouterException("Only chunked uploads can have reserved metadata blob id passed in.",
+            RouterErrorCode.InvalidOrMismatchedStitchBlobReservedMetadataChunkId);
+      }
+      if (options.isChunkUpload() && passedInBlobProperties.getReservedMetadataBlobId() == null) {
+        reservedMetadataIdMetrics.noReservedMetadataForChunkedUploadCount.inc();
+        throw new RouterException("No reserved metadata passed in for chunked upload",
+            RouterErrorCode.InvalidOrMismatchedStitchBlobReservedMetadataChunkId);
+      }
+    }
+
+    /**
+     * Validate and set the specified chunkId as the reserved metadata chunk id.
+     * @param chunkId The chunk id to set.
+     * @throws RouterException In case of any issue during validation.
+     */
+    private void setReservedMetadataChunkId(String chunkId) throws RouterException {
+      try {
+        reservedMetadataChunkId = new BlobId(chunkId, clusterMap);
+      } catch (Exception exception) {
+        reservedMetadataIdMetrics.stitchedBlobMetadataIdDeserErrorCount.inc();
+        throw new RouterException("Invalid metadata chunk id of chunks in the stitched operation.",
+            RouterErrorCode.InvalidOrMismatchedStitchBlobReservedMetadataChunkId);
+      }
+      if (reservedMetadataChunkId.getBlobDataType() != BlobDataType.METADATA) {
+        reservedMetadataIdMetrics.stitchedBlobMetadataIdDeserErrorCount.inc();
+        throw new RouterException("The blob data type of the metadata blob should be BlobIdType.METADATA",
+            RouterErrorCode.InvalidOrMismatchedStitchBlobReservedMetadataChunkId);
+      }
+    }
+
+    /**
+     * Verify that the reserved metadata chunk id matches the specified chunk id.
+     * @param chunkId chunk id to be matched.
+     * @throws RouterException In case of a mismatch.
+     */
+    private void verifyStitchedBlobReservedMetadataChunkId(String chunkId) throws RouterException {
+      if (reservedMetadataChunkId == null && chunkId == null) {
```

Comment:
As you mentioned we provide the reserved meta data id already even the flag is not enable, right?
If so, this is an error case, we shouldn't ignore, right?

Reply:
Actually we don't need to handle this case separately. This will already be handled by the code before this and in this method.